### PR TITLE
Add binary data transfer from .net extensions, as per XWALK-4615.

### PR DESCRIPTION
**Pull request by zcr01**

### CODE REVIEW

**Comment by darktears** [comment]
Thanks for the patch. You should probably rebase your patch to the current master (I can see a Merge branch 'master' of https://github.com/zcr01/crosswalk into HEAD")

Also it's missing tests.

**Comment by zcr01** [comment]
Tests are here: https://github.com/crosswalk-project/crosswalk-test-suite/pull/3847

**Comment by darktears** [comment]
It should be there : https://github.com/crosswalk-project/crosswalk/tree/master/extensions/test/win

**Comment by zcr01** [comment]
Ok - I'll put tests over there, probably tomorrow at this point.  Sorry, first pr and the contribute page directed me to the other repository.

**Comment by zcr01** [comment]
Added tests and cleaned the history.

**Comment by darktears** [comment]
Could you please just reformat the commit message to :

[Windows] Add binary data transfer for .net extensions
Explain the commit here.

BUG=XWALK-4615

**Comment by darktears** [comment]
LGTM. I'll merge when the bots are green. Thanks for the patch.

**Comment by rakuco** [comment]
@darktears @zcr01: I think this broke our official Windows canary builds:

```
[11955/17412] ACTION //xwalk/extensions/test/win:dotnet_binary_extension__msbuild(//build/toolchain/win:x64)
FAILED: tests/dotnet_extension/binary_extension/binary_extension.dll 
C:/Users/blduser/workdir/src/depot_tools/python276_bin/python.exe ../../xwalk/tools/msbuild_dotnet.py --build-type Release --environment-file environment.x64 --project ../../xwalk/extensio
ns/test/win/binary_extension/binary_extension.csproj --output-dir tests/dotnet_extension/binary_extension
XWalkExtension.cs(20,7): error CS1039: Unterminated string literal [C:\Users\blduser\workdir\src\src\xwalk\extensions\test\win\binary_extension\binary_extension.csproj]
XWalkExtension.cs(33,1): error CS1002: ; expected [C:\Users\blduser\workdir\src\src\xwalk\extensions\test\win\binary_extension\binary_extension.csproj]
XWalkExtension.cs(33,1): error CS1513: } expected [C:\Users\blduser\workdir\src\src\xwalk\extensions\test\win\binary_extension\binary_extension.csproj]
Traceback (most recent call last):
  File "../../xwalk/tools/msbuild_dotnet.py", line 70, in <module>
    sys.exit(main())
  File "../../xwalk/tools/msbuild_dotnet.py", line 66, in main
    LaunchMSBuild(options)
  File "../../xwalk/tools/msbuild_dotnet.py", line 48, in LaunchMSBuild
    subprocess.check_call(cmd)
  File "C:\Users\blduser\workdir\src\depot_tools\python276_bin\lib\subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['msbuild', '/nologo', '/p:Platform=AnyCPU,Configuration=Release,OutDir=C:\\Users\\blduser\\workdir\\src\\src\\out\\Release\\tests\\dotnet_extension
\\binary_extension,BaseIntermediateOutputPath=C:\\Users\\blduser\\workdir\\src\\src\\out\\Release\\tests\\dotnet_extension\\binary_extension\\', '/verbosity:quiet', '../../xwalk/extensions
/test/win/binary_extension/binary_extension.csproj']' returned non-zero exit status 1
```

File: extensions/test/win/binary_extension/XWalkExtension.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      @"var binaryTestListener = null;
+        extension.setMessageListener(function(msg) {
+          if (binaryTestListener instanceof Function) {
+            binaryTestListener(msg);
+          };
+        });
+        exports.binaryEcho = function(msg, callback) {
+          binaryTestListener = callback;
+          extension.postMessage(msg);
+        };
```
Comment (darktears) [inline extensions/test/win/binary_extension/XWalkExtension.cs]: It's missing a " here.

**Comment by zcr01** [comment]
Should be fixed now.  I think I must have messed something up in the rebase and missed the commit where I fixed that.

**Pull request merged**